lib/spec/sww.sw-spec.js

```diff
@@ -0,0 +1,71 @@
+importScripts('/base/lib/spec/nodeMock.js');
+importScripts('/base/lib/sww.js');
+importScripts('/base/lib/router.js');
+
+describe('ServiceWorkerWare', function () {
+  'use strict';
+
+  var worker;
+  var request;
+
+  var fallbackMW = function(req, res) {
```

Comment:
Then, you are not ensuring the semantics of the default middleware: it's only called if no other has been hit before because this depend on the implementation of the middleware itself. You have two options here:
  * Relax the definition of the bug to simply this: default is the last middleware called.
  * Decorate the fallback with this `res !== null` check to ensure the default is the middleware that runs at the end if no other middleware has answered before.